pkg/api/response/response.go

```diff
@@ -292,6 +296,10 @@ func ErrOrFallback(status int, message string, err error) *NormalResponse {
 		return Err(err)
 	}
 
+	if errors.Is(err, context.Canceled) {
+		return Err(errRequestCanceledBase.Errorf("response: request canceled: %w", err))
+	}
+
```

Comment:
This feels like a hack, but it seems like the easiest way to catch any `context.Canceled` errors being returned to the API layer. WDYT?

Reply:
I agree it kind of does but it also makes sense. I think it's good, but just to be sure I would also change the docstring so it's clear that if an error wrapping `context.Canceled` is provided, the function returns an `Err(errRequestCanceledBase)`

Reply:
Good point. I'll have a look